json/src/main/java/org/teavm/flavour/json/deserializer/ArrayDeserializer.java

```diff
@@ -30,7 +30,7 @@ public ArrayDeserializer(Class<?> itemType, JsonDeserializer itemDeserializer) {
 
     @Override
     public Object deserialize(JsonDeserializerContext context, Node node) {
-        if (node.isNull()) {
+        if (node.isMissing() || node.isNull()) {
```

Comment:
I don't understand why you add this 'isMissing' check to every type of deserializer instead of checking at at property deserialization level. Also, missing != null, they are different and must induce different behaviour.

Reply:
I understand that missing is undefined (that corresponds to undeclared field in JSON). Currently to avoid issues, I need to write "property": null to nullable properties. Jackson handles missing properties well, TeaVM deserializer don't. So this is the reason why I added this isMissing check. Actualy it can be simplified to value == null, but I didn't want to introduce non-standard method in JSON model.
My idea is that int value should fail during deserialization, but Integer value shouldn't because when it's undeclared and I emphasized that it's nullable. Currently it fails and forces me to put lots of "property": null redundant declarations into my model.

Reply:
This does not answer my question. I can't approve this PR until you resolve the issue I'm trying to address.

Reply:
I added it near isNull check. So as I understand, you're suggesting to move it out deserializers and remove all node.isMissing and node.isNull checks in deserializers?

Reply:
Not exactly. 'null' check is ok here, because 'null' is valid JSON value. And 'null' in JSON *ALWAYS* means 'null' in Java, while missing property does not always mean 'null' in Java.

Reply:
Someone already tried to fix this issue, but his patch was incomplete, nor it included any tests.